### ISSUE TITLE
Random Freezes of the mouse, impossible to play

### ISSUE THREAD

**Issue by Ayfri** [ppy/osu#11913]
The mouse freeze at random places randomly making the game impossible to play...

**Screenshots or videos showing the encountered issue:** 
[Youtube Clip](https://youtu.be/etJM0yFFP8c)

**osu!lazer version:** 
2021.226.0

**Logs:**

[performance.log](https://github.com/ppy/osu/files/6053442/performance.log)
[runtime.log](https://github.com/ppy/osu/files/6053443/runtime.log)
[updater.log](https://github.com/ppy/osu/files/6053444/updater.log)
[database.log](https://github.com/ppy/osu/files/6053445/database.log)
[network.log](https://github.com/ppy/osu/files/6053446/network.log)

**Computer Specifications:** 
RAM: 16 Go
GPU: NVidia GTX 1660 Super
CPU: AMD Ryzen 5 3600 6-Core

**Comment by bdach** [comment]
What input device are you using to reproduce this? Do you have any custom/non-vendor device drivers installed?

**Comment by Ayfri** [comment]
No, my mouse is a Logitech G502, I'm using the software from Logitech, it is up to date, etc.

**Comment by bdach** [comment]
We're going to need another video of this then, but with the extended FPS counter showing (<kbd>Ctrl</kbd>+<kbd>F11</kbd> twice).

**Comment by Ayfri** [comment]
Okay, I'm not at home right now so you will get it in about 6-7 hours ^^

**Comment by Ayfri** [comment]
Okay sorry for taking so long I was really tired yesterday, [here it is](https://youtu.be/ohWX4SsDNPk).

**Comment by bdach** [comment]
I honestly can't see anything that would indicate that this is the game's fault. Were you moving the mouse throughout the entire video?

The input thread doesn't seem to lag. All input frames seem to be within expected range (<1ms). At the end of the video when the mouse stops, it goes to 0ms, which may indicate a freeze, but it may as well just indicate no input. The mouse also seems to slightly move when "stuck".

In other words: are you sure this isn't just hardware failing on you? Is lazer the *only* app this happens in?

**Comment by Ayfri** [comment]
Yes, it is the only one with freeze, and on 2021.220.0 I wasn't getting any lags, 60-61 fps all the time. I'll try to make a new video showing more explicitly the lag, it happens a lot when playing so I guess I'll play ~~(with my horrible level)~~.

**Comment by bdach** [comment]
> and on 2021.220.0 I wasn't getting any lags, 60-61 fps all the time

This is very weird. Can you double-confirm this by downgrading and checking again? I don't really understand how that could happen. If anything, the last update should have made performance *better*, and not worse.

**Comment by Ayfri** [comment]
I suppose that I can't downgrade into the app (like the updates) ?

**Comment by bdach** [comment]
Previous binaries should be still there, in `%LocalAppData%/osulazer/app-${VERSION}`, I believe.

**Comment by Ayfri** [comment]
Oh found it thanks!

**Comment by Joehuu** [comment]
I think this is the same as https://github.com/ppy/osu/issues/10780.

**Comment by Ayfri** [comment]
Oh yes, this is possible, I'm using 1000Hz for my mouse.

**Comment by bdach** [comment]
I'm not too sure about that. I'd be expecting the spikes to *not* be gone during the freeze time. Could try dropping the polling rate below 1000Hz, though, if even just for testing.

**Comment by peppy** [comment]
I use the same mouse and same software, so it's safe to say it's not related to the hardware here. There must be something going weird with our polling / interpolation / something under certain performance scenarios.

**Comment by peppy** [comment]
@bdach i'm not sure if you missed it, but in their video their input thread has some insanely huge spikes in `WndProc` suggesting that something else on their system is taking time to process their mouse input.

@Ayfri could you please check for other applications running on your system that may be intercepting your mouse? make sure to close anything which shows a graphical overlay on your game, or anything that could be interacting with your mouse (basically close everything).

**Comment by Ayfri** [comment]
When  recording, only Discord, OBS & Osu!Lazer was open, I don't think that OBS nor Discord can do this (and it is litteraly the first time that I have a bug like this in Osu!Lazer, even in Osu!).

I'm not at home for 4 until 4 days so I can't record another video ^^'

**Comment by LeNitrous** [comment]
This should probably be moved to osu-framework. `SampleGame` and our project is also affected by this. Something to note, I use a Logitech G102 and have the Logitech G Hub installed. I have tested by closing all possible overlays (e.g. Windows Game Mode Overlay, Discord, NVIDIA Shadowplay) that it persists.

**Comment by peppy** [comment]
@LeNitrous since you are (presumable) able to debug, can you do so and find the cause? I also use logitech hardware and cannot reproduce. Maybe start by uninstalling the software?

**Comment by LeNitrous** [comment]
I'm not very well versed into debugging these kinds of things so I can't really give any other help other than trying out ideas however I did attempt uninstalling the software and unfortunately the issue still persists.

**Comment by paz** [comment]
i'm getting freezes with a g502 aswell, however on the original osu version I have no problems.

**Comment by Susko3** [comment]
@LeNitrous To try and narrow down the issue, I suggest you try to [perform a clean boot in Windows](https://support.microsoft.com/en-us/topic/how-to-perform-a-clean-boot-in-windows-da2f9573-6eec-00ad-2f8a-a97a1807f3dd).

If the issue is not present with a clean boot: Then try with the services (you disabled for clean boot) enabled, but keep startup programs disabled.

Try to narrow it down to a single program or service.

**Comment by peppy** [comment]
Has anyone tried this one without the logitech software running yet? Would be good to get to the bottom of it.

**Comment by salmanmkc** [comment]
I have lag issues with the original version of osu with 1000hz on the G305, 500Hz there's a bit but 250 is much better

**Comment by LeNitrous** [comment]
I've done further investigation and it possibly only happens on Windows (I don't have other operating systems to test on) as many other games (it also happens on stable) have also had this issue with the common denominator being that they also had 1000hz polling rate on their devices. 

![image](https://user-images.githubusercontent.com/20495991/126084758-248fb814-ba17-4dff-b40b-dd45de3b5a44.png)
![image](https://user-images.githubusercontent.com/20495991/126084764-7992a42e-0062-44ad-986a-1a9040aa92a1.png)

The solution as @salmanmkc proposed is to turn it down to the lowest possible setting which in many cases of other people is to set it to 125hz. Unfortunately I also don't have the luxury to check which version of Windows 10 this started to cause issues with but a hint is when around the time they made changes to touch screen and tablet input.

**Comment by peppy** [comment]
I'm going to go out on all my limbs here and say that this *is not a lazer issue*, but involves something running on your PC that is handling the input events and causing a high overhead, external to the game you think is the issue. Also I can guarantee to you that it is *not* "windows" at fault. It is going to be some software running on your system. Not windows.

Please test at windows desktop with task manager up. Figure what is consuming your CPU.

Here are some tips on how to diagnose your own performance issues: https://osu.ppy.sh/help/wiki/Performance_Troubleshooting 

I'm going to close this thread because the same advice applies to the opening post as well.
